lib/evmone/eof.cpp

```diff
@@ -98,15 +100,46 @@ std::pair<EOFSectionHeaders, EOFValidationErrror> validate_eof_headers(bytes_vie
     return {section_headers, EOFValidationErrror::success};
 }
 
-std::pair<EOF1Header, EOFValidationErrror> validate_eof1(bytes_view code) noexcept
+EOFValidationErrror validate_instructions(evmc_revision rev, bytes_view code) noexcept
 {
-    const auto [section_headers, error] = validate_eof_headers(code);
-    if (error != EOFValidationErrror::success)
-        return {{}, error};
+    assert(code.size() > 0);  // guaranteed by EOF headers validation
+
+    size_t i = 0;
+    uint8_t op;
+    do
+    {
+        op = code[i];
+        const auto& since = instr::traits[op].since;
+        if (!since.has_value() || *since > rev)
```

Comment:
I think this doesn't cover the case when we want to remove an opcode (e.g.`calldata`) from EOF contracts?

Reply:
Yes, when that happens I guess we will need to add final revision to `traits`.

Reply:
Yes. This looks like a good solution if needed.